apollo-router/src/services/supergraph_service.rs

```diff
@@ -245,15 +247,27 @@ async fn plan_query(
 }
 
 fn accepts_multipart(headers: &HeaderMap) -> bool {
-    let multipart_mixed = MediaType::new(MULTIPART, MIXED);
-
     headers.get_all(ACCEPT).iter().any(|value| {
         value
             .to_str()
             .map(|accept_str| {
                 let mut list = MediaTypeList::new(accept_str);
 
-                list.any(|mime| mime.as_ref() == Ok(&multipart_mixed))
+                list.any(|mime| {
+                    mime.as_ref()
+                        .map(|mime| {
+                            mime.ty == MULTIPART
+                                && mime.subty == MIXED
```

Comment:
FWIW my impression is that this particular library separates out "the thing after the +" so this woudl also match `multipart/mixed+foo` which is probably not ideal. I would suggest just doing `mime.essence() == multipartmixed` (essence strips parameters)